lib/sendcloud/errors/response_error.rb

```diff
@@ -0,0 +1,21 @@
+module Sendcloud
+  module Errors
+    class ResponseError < StandardError
+      attr_reader :payload, :body, :status
+
+      def initialize(payload:, body:, status:)
+        @payload = payload
+        @body = body
+        @status = status
+
+        super(build_message)
+      end
```

Comment:
Updated this to be consistent with DHL Express. Status code now comes from the response.